benchmarks/scripts/plot_avg_benchmarks_over_time.py

```diff
@@ -151,8 +166,8 @@ def extract_compiler_versions(header):
     )
 
 # Add annotations for version changes
-for date in avg_compile_time["date"].unique():
-    date_data = avg_compile_time[avg_compile_time["date"] == date]
+for date in filtered_avg_compile_time["date"].unique():
+    date_data = filtered_avg_compile_time[avg_compile_time["date"] == date]
```

Comment:
I think this is mixing dataframes, consider:
```suggestion
    date_data = filtered_avg_compile_time[filtered_avg_compile_time["date"] == date]
```